### ISSUE TITLE
False negative ObjectPropertyNaming in RC11 

### ISSUE THREAD

**Issue by scottkennedy** [arturbosch/detekt#1346]
My config has:

```yml
ObjectPropertyNaming:
  privatePropertyPattern: '^(?![ms][A-Z])(([xyz]|[a-z][a-zA-Z0-9]|_[a-z])[a-zA-Z0-9]*)$'
```

I have code with

```kt
companion object {
    private val SESSION = Session()
}
```

This should not be allowed, as my config clearly does not want ALL_CAPS for private properties.

## Expected Behavior
`Private object property names should match the pattern: ^(?![ms][A-Z])(([xyz]|[a-z][a-zA-Z0-9]|_[a-z])[a-zA-Z0-9]*)$`

## Observed Behavior
No issue reported.

## Context
This worked as expected in RC10

## Your Environment
* Version of detekt used: RC11
* Version of Gradle used (if applicable): 4.10.2
* Operating System and version: Various

**Comment by schalkms** [comment]
I couldn't reproduce this issue. It gets reported on my machine.

**Comment by scottkennedy** [comment]
Ah I think it's caused by #1345.  Something fails, so it stops running on the class, and therefore doesn't report it.  I've had to turn off `UnusedPrivateClass` due to #1347 anyway, so that "fixes" my issue.
